shell/components/Resource/Detail/Card/PodsCard/composable.ts

```diff
@@ -0,0 +1,32 @@
+import { useI18n } from '@shell/composables/useI18n';
+import { computed, toValue } from 'vue';
+import { useStore } from 'vuex';
+
+export type ScaleFn = () => Promise<void>;
+export const useScaling = (workloadName: string, scaleUpFn: ScaleFn, scaleDownFn: ScaleFn) => {
+  const store = useStore();
+  const i18n = useI18n(store);
+
+  const scale = (scaleFn: () => Promise<void>, workloadName: string, direction: 'up' | 'down') => {
+    return async() => {
+      try {
+        await scaleFn();
+      } catch (err) {
+        store.dispatch('growl/fromError', {
+          title: i18n.t('workload.list.errorCannotScale', { workloadName, direction }),
+          err
+        },
+        { root: true });
+      }
+    };
+  };
+
+  const workloadNameValue = toValue(workloadName);
+  const scaleUp = computed(() => scale(() => scaleUpFn(), workloadNameValue, 'up'));
+  const scaleDown = computed(() => scale(() => scaleDownFn(), workloadNameValue, 'down'));
```

Comment:
What do we gain out of `scaleUp` and `scaleDown` being computed properties? It looks like both of these would be better suited as methods, without the need for caching.

Reply:
Yeah, you're right. I've switched them to methods. I just got too computed happy after some reactivity issue chasing.